src/background.js

```diff
@@ -29,50 +29,74 @@ function sendVideoUrlToMetubeAndSwitchTab(videoUrl, metubeUrl, tab) {
+                throw new Error('Error when adding URL to MyTube');
+            }
         }
-    }).catch(function(res) {
-        alert("error :: " + res);
+    }).then(function() {
+        typeof callback === 'function' && callback();
+    }).catch(function(e) {
+        console.log("Ran into an error :: ", e);
     });
 }
 
 chrome.contextMenus.onClicked.addListener(function(item, tab) {
-    chrome.storage.sync.get(['metube'], function(data) {
+    chrome.storage.sync.get(['metube', 'contextMenuClickBehavior'], function(data) {
         if (data === undefined || !data.hasOwnProperty('metube') || data.metube === "") {
             openTab(chrome.runtime.getURL('options.html'), tab);
             return
         }
-        sendVideoUrlToMetubeAndSwitchTab(item.linkUrl, data.metube, tab);
+        if ( data.contextMenuClickBehavior == 'context-menu-send-current-url') {
+            sendVideoUrlToMetube(item.linkUrl, data.metube);
+        } else if (data.contextMenuClickBehavior == 'context-menu-send-current-url-and-switch'){
+            sendVideoUrlToMetube(item.linkUrl, data.metube, function() {
+                openTab(data.metube, tab);
+            });
+        } else {
+            console.log("Unknown contextMenuClickBehavior value: " + data.contextMenuClickBehavior);
+        }
     });
 });
 
 chrome.action.onClicked.addListener(function(tab) {
-    chrome.storage.sync.get(['metube', 'sendOnClick'], function(data) {
+    chrome.storage.sync.get(['metube', 'clickBehavior'], function(data) {
         if (data === undefined || !data.hasOwnProperty('metube') || data.metube === "") {
             openTab(chrome.runtime.getURL('options.html'), tab);
             return
         }
-        chrome.tabs.query({
-            active: true,
-            lastFocusedWindow: true
-        }, function(tabs) {
-            // use this tab to get the youtube video URL
-            let videoUrl = tabs[0].url;
-            if (data.sendOnClick) {
-                sendVideoUrlToMetubeAndSwitchTab(videoUrl, data.metube, tab);
-            }
-            else {
-                console.log("Going to Metube URL...");
-                openTab(data.metube, tab);
-            }
-        });
+        if (data.clickBehavior == 'do-nothing') {
+            return
+        } else if (data.clickBehavior == 'go-to-metube') {
+            console.log("Going to Metube URL...");
+            openTab(data.metube, tab);
+        } else if (data.clickBehavior == 'send-current-url') {
+            chrome.tabs.query({
+                active: true,
+                lastFocusedWindow: true
+            }, function(tabs) {
+                // use this tab to get the youtube video URL
+                let videoUrl = tabs[0].url;
+                sendVideoUrlToMetube(videoUrl, data.metube);
+            });
+        } else if (data.clickBehavior == 'send-current-url-and-switch') {
+            chrome.tabs.query({
```

Comment:
you duplicate that block of code but difference only on `openTab` callback. i think you can extract it to another function and not duplicate code.

Reply:
The videoUrl is only available in the callback function scope in the chrome.tabs.query function.
I tried to make a single function to fetch the videoUrl and then call it twice, but ultimately gave up as I couldn't make it work.

Perhaps you can suggest how I could extract common code from here as a function, and I'm eager to learn here :)

Reply:
i not tested that code, but i think you can start from something like that

```javascript
chrome.action.onClicked.addListener(function(tab) {
    chrome.storage.sync.get(['metube', 'clickBehavior'], function(data) {
        if (data === undefined || !data.hasOwnProperty('metube') || data.metube === "") {
            openTab(chrome.runtime.getURL('options.html'), tab);
            return;
        }

        if (data.clickBehavior == 'go-to-metube') {
            openTab(data.metube, tab);
            return;
        }

        let needToSwitch = (data.contextMenuClickBehavior === 'context-menu-send-current-url-and-switch');

        sendVideoUrlToMetube(tabs[0].url, data.metube, function() {
            if(!needToSwitch){
                return;
            }
            openTab(data.metube, tab);
        });
    });
});

```

Reply:
Done - added needToSwitch to simplify code